python/cudf/cudf/core/column/decimal.py

```diff
@@ -364,18 +364,51 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     else:
         raise NotImplementedError()
 
+    if isinstance(lhs_dtype, type(rhs_dtype)):
+        # SCENARIO 1: If `lhs_dtype` & `rhs_dtype` are same, then try to
+        # see if `precision` & `scale` can be fit into this type.
+        try:
+            return lhs_dtype.__class__(precision=precision, scale=scale)
+        except ValueError:
+            # Call to _validate fails, which means we need
+            # to goto SCENARIO 3.
+            pass
+    else:
+        # SCENARIO 2: If `lhs_dtype` & `rhs_dtype` are of different dtypes,
+        # then try to see if `precision` & `scale` can be fit into the type
+        # with greater MAX_PRECISION (i.e., the bigger dtype).
+        try:
+            if lhs_dtype.MAX_PRECISION > rhs_dtype.MAX_PRECISION:
```

Comment:
I think this is effectively the same, but `>=` might be clearer in its intent, rather than having `==` fall through to the `else` block.
```suggestion
            if lhs_dtype.MAX_PRECISION >= rhs_dtype.MAX_PRECISION:
```